pkg/querier/tripperware/query_attribute_matcher.go

```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/cortexproject/cortex/pkg/util/validation"
 )
 
-const QueryRejectErrorMessage = "This query has been rejected by the service operator."
+const QueryRejectErrorMessage = "This query does not perform well and has been rejected by the service operator."
```

Comment:
It is probably unnecessary for this one. But we do make the message a flag in Ingester in order to customize it. https://github.com/cortexproject/cortex/blob/master/pkg/ingester/ingester.go#L162

Reply:
I wonder if we want to include the first part `does not perform well`. If a regex accidentally matches a good query, does it confuse users?

Reply:
In that case, it gives good reason to user to reach out service operator asking why that query does not perform well. We want user to reach out to service team if query is rejected by accident so I think it's good idea to have this part of the error message in case of accidental match.

Reply:
OK. I think we can leave it for now.